### PR TITLE
Ensure schemas are kept up-to-date

### PR DESCRIPTION
**Pull request by brennie**
This adds a script `fetch-schemas.py` to sync our in-tree schemas with mozilla-central. It also adds a Circle CI check to make sure we do not merge with out-of-date schemas.

### CODE REVIEW

**Comment by brennie** [comment]
I intentionally did not sync the schemas and commit them to ensure that the Circle CI step worked.

**Comment by brennie** [comment]
The check will fail CircleCI because the schema changes from https://phabricator.services.mozilla.com/D146028 have not landed yet and were imported manually.

Locally, `make check` validates all messages successfully.

**Comment by brennie** [comment]
This currently only validates against the FxMS Mega Schema for messages (instead of individual message template schemas) as described in the last commit message.

**Comment by brennie** [comment]
@punamdahiya I was not able to request review from you.

File: schema/message-groups.schema.json [hunk 1]
```diff
@@ -52,12 +56,8 @@
           }
         }
       }
-    },
-    "type": {
-      "type": "string",
-      "description": "Local auto-generated group or remote group configuration from RS.",
-      "enum": ["remote-settings", "local", "default"]
```
Comment (brennie) [inline schema/message-groups.schema.json:59]: This field is still present, so we should update this in m-c to re-add this field 

https://searchfox.org/mozilla-central/rev/2bbb0c0a90df20702df8c8011a8996536a83cb75/browser/app/profile/firefox.js#1535-1542

File: schema/message-groups.schema.json [hunk 2]
```diff
@@ -38,8 +38,12 @@
...
             "properties": {
               "period": {
-                "type": "integer",
-                "description": "Period of time in milliseconds (e.g. 86400000 for one day)"
+                "oneOf": [
+                  {
+                    "type": "integer",
+                    "description": "Period of time in milliseconds (e.g. 86400000 for one day)"
+                  }
+                ]
```
Comment (brennie) [inline schema/message-groups.schema.json:46]: This is wacky and can be undone in m-c

File: scripts/validate.py [hunk 3]
```diff
@@ -1,74 +1,95 @@
...
 
-        validate.py ${TYPE} ${JSON_PATH}
-
-    Where ${type} should be one of "cfr" "cfr-fxa", "messaging-experiments",
-    "whats-new-panel".
+    Where TYPE should be one of "message", "experiment", or "message-groups".
 
     Exmaple:
-        validate.py cfr ./cfr.json
+        validate.py message ./cfr.json
```
Comment (brennie) [inline scripts/validate.py:65]: This should be `./outgoing/cfr.json`

File: scripts/validate.py [hunk 4]
```diff
@@ -101,8 +118,11 @@ def validate_action(action, for_exp):
         sys.exit(1)
 
 
-def extract_actions(message, message_type):
-    def _extract_cfr():
+def extract_actions(message):
+    """Extract all of the special message actions from the given message."""
+    message_type = message["template"]
+
+    def _extract_cfr_doorhanger():
```
Comment (brennie) [inline scripts/validate.py:125]: this should be `_extract_extension_doorhanger()`

**Review by AllegroFox — APPROVED** [review]
LGTM with nits

File: schema/message-groups.schema.json (hunk 2 above)
Comment (brennie) [reply schema/message-groups.schema.json:46]: Waiting on https://bugzilla.mozilla.org/show_bug.cgi?id=1782113 to merge to m-c

File: schema/message-groups.schema.json (hunk 1 above)
Comment (brennie) [reply schema/message-groups.schema.json:59]: Waiting on https://bugzilla.mozilla.org/show_bug.cgi?id=1782113 to merge to m-c

**Pull request merged**